pandas/core/reshape/pivot.py

```diff
@@ -424,42 +425,48 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, Sequence[Optional[Label]]]] = None,
```

Comment:
does `Optional[Label]` mean something distinct from `Label`?

Reply:
what i want to express here is that it could be an empty `[]`, so i put an `Optional` here, maybe it's already covered in `Label`? @jbrockmendel

Reply:
yes, `Label` is defined as `Optional[Hashable]`

Reply:
i t hink would be ok to add these to _typing, maybe Labels? so this becomes `Optional[Labels]`

Reply:
yeah there's a difference between Optional used for keyword args and Optional added to Labels to cover None since Hashable doesn't include None.

In general, an alias in typing would not include Optional.

so `Optional[Labels]` would be correct and `Labels` would be defined accordingly. I _think_ `Union[Label, Sequence[Label]]` would be sufficient.

empty is different again.

```
from typing import List

reveal_type([])  # Revealed type is 'builtins.list[<nothing>]'

reveal_type([1, 2])  # Revealed type is 'builtins.list[builtins.int*]'

reveal_type([1, None])  # Revealed type is 'builtins.list[Union[builtins.int, None]]'

a: List[int] = []  # no errors
```
where `Union[builtins.int, None]` == `Optional[int]`

so `Optional` is different from empty. `Optional[<type>]` is syntatic sugar for `Union[<type>, None]`. An empty list is compatible with lists of any type.

In mypy, I don't think it is possible to define a type as an empty list (it is possible for a tuple)

from https://www.python.org/dev/peps/pep-0484/#the-typing-module

> The empty tuple can be typed as Tuple[()]